### PR TITLE
feat: implement `plan/get` capability

### PR DESCRIPTION
**Pull request by travis**
In console, we need a way to tell if a user has a subscription. Implement the `plan/get` capability from https://github.com/web3-storage/w3up/issues/959 to enable that.

Includes the core capability definitions plus `upload-api` handlers and `access-client` helper functions.

### CODE REVIEW

File: packages/access-client/test/agent-use-cases.test.js [hunk 1]
```diff
@@ -188,3 +192,50 @@ describe('authorizeWaitAndClaim', async function () {
...
+  await Promise.all([
+    ...await createAuthorization({ account: accountWithAPlan, agent: agent.issuer, service: server.id }),
+    ...await createAuthorization({ account: accountWithoutAPlan, agent: agent.issuer, service: server.id })
+  ].map(proof => agent.addProof(proof)))
+
+  it("should succeed for accounts with plans", async function () {
+    assert((await getAccountPlan(agent, accountWithAPlan)).ok)
+  })
+
+  it("should fail for accounts without a plan", async function () {
```
Comment (gobengo) [inline packages/access-client/test/agent-use-cases.test.js:238]: could be nice to assert the error here. i.e. if I am a client and I try to plan/get, and there is an error, how do I know whether the error is a sign that my account has no plan (but a UI could still help user set one) vs a sign of something more wrong in the system (and e.g. a UI should show a 'come back later' state if the error is unknown).

Comment (gobengo) [reply packages/access-client/test/agent-use-cases.test.js:238]: e.g. I think this would be a PlanNotFound ?

File: packages/capabilities/test/capabilities/plan.test.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    const result = await access(await auth.delegate(), {
+      capability: Plan.get,
+      principal: Verifier,
+      authority: service,
+      validateAuthorization,
+    })
+    assert.equal(result.error?.message.includes('not authorized'), true)
+  })
+
+  it('can delegate plan/add', async function () {
```
Comment (gobengo) [inline packages/capabilities/test/capabilities/plan.test.js:68]:
```suggestion
  it('can delegate plan/get', async function () {
```

File: packages/upload-api/test/handlers/plan.js
Comment (gobengo) [inline packages/upload-api/test/handlers/plan.js:1]: Is this file really testing anything 'plan' related? I expected it to be based on the filename being `plan.js`

File: packages/upload-api/test/handlers/plan.spec.js
Comment (gobengo) [inline packages/upload-api/test/handlers/plan.spec.js:1]: Is this rename intentional? The file contents don't mention plans

Comment (travis) [reply packages/upload-api/test/handlers/plan.spec.js:1]: oh jeez, copy/paste fail, thx!

File: packages/capabilities/src/types.ts [hunk 3]
```diff
@@ -514,6 +515,21 @@ export type AggregateAccept = InferInvokedCapability<
   typeof DealerCaps.aggregateAccept
 >
 export type DealInfo = InferInvokedCapability<typeof DealTrackerCaps.dealInfo>
+
+// Plan
+
+export type PlanGet = InferInvokedCapability<typeof PlanCaps.get>
+export interface PlanGetSuccess {
+  updatedAt: string
```
Comment (gobengo) [inline packages/capabilities/src/types.ts:523]: What kind of string? Reading the code it looks like this will usually be an ISO8601 datetime string.
* I think that's a great choice
* notably, it is different than the number used in https://github.com/web3-storage/w3up/issues/959 . But again, I prefer a string to a number here
* string/number aside, here in the `PlanGetSuccess` typescript type, IMO it's worth adding a comment explaining constraints like ISO8601 string
  * may also be testing this in relevant tests that receive a PlanGetSuccess, i.e. can the PlanGetSuccess#updatedAt value be parsed as ISO8601 to a realistic datetime

File: packages/upload-api/test/handlers/plan.spec.js
Comment (travis) [reply packages/upload-api/test/handlers/plan.spec.js:1]: ok this is fixed - new test added

File: packages/access-client/test/agent-use-cases.test.js (hunk 1 above)
Comment (travis) [reply packages/access-client/test/agent-use-cases.test.js]: great idea - done

File: packages/capabilities/src/types.ts (hunk 3 above)
Comment (travis) [reply packages/capabilities/src/types.ts]: ah good call, agreed! I've left a comment on #959 to this effect, and have created a new `string` alias called `ISO8601Date` so that this is clearer at least. also added a test to the handler that verifies we keep dates in 8601 format throughout the stack.

**Review by gobengo — APPROVED** [review]
Looks like recent commits incorporated all my first wave of feedback
